src/Sarif/PropertiesDictionaryExtensionMethods.cs

```diff
@@ -98,7 +98,7 @@ public static class PropertiesDictionaryExtensionMethods
                 IDictionary pb = property as IDictionary;
                 if (pb != null)
                 {
-                    ((IDictionary)pb).SavePropertiesToXmlStream(writer, settings, key, settingNameToDescriptionMap);
+                    pb.SavePropertiesToXmlStream(writer, settings, key, settingNameToDescriptionMap);
```

Comment:
>SavePropertiesToXmlStream [](http://example.com/codeflow?start=23&length=25)

Some of the code changes clearly have no potential for runtime impact and we therefore have a high degree of confidence in taking them. This is one because pb is explicitly typed as IDictionary (and there are others in your change). What we need to look for are situations where someone may have overloaded a cast operator, or where a change you made may have changed precedence ordering/evaluation.

One thing you should do here is to generate code coverage and see where we have coverage. If we do not have coverage in a place where the change looks like it has potential to change tool behavior, we need new tests.

You should always describe why you are not submitting a test, if you submit a pull request without them. @eddynake, your help providing this feedback in a review context would be appreciated as well.

Reply:
OK, I reviewed this carefully and have convinced myself there are no behavioral implications of this change, so I won't block on testing. Please do bear in mind what I said: if you have no tests, please tell the reviewers why you believe they are not required. Core expectation.

Reply:
Agree with the inputs here, sorry did not mention anything.
Just added in the top description why tests are not authored together with the change.